### ISSUE TITLE
Support for offset and limit

### ISSUE THREAD

**Issue by alexander-schranz** [loupe-php/loupe#13]
Currently it is build for only paginated views. But it would be great if also offset and limits would work. Some usecase for this e.g.:

Display first the Top 10 matches but have a button showAll which will then `offset: 10` without any `limit`.

My suggestion is to use in `SearchParameters` `limit` and `offset` but have 2 methods e.g.:

 - `withPagination(int $page, int $hitsPerPage);`
 - `withLimitOffset(?int $limit, int $offset);`

**Comment by Toflar** [comment]
I have yet to see a use case for `offset` and `limit`. To me, `page` and `hitsPerPage` is so much more intuitive.

I also don't think that "without any limit" exists in reality. There is always a limit. Either you define a limit by configuring it or - if we would allow to configure it to `null` or `PHP_INT_MAX` -  Loupe crashes because there's too much data to fetch. This is also a limit. And crashing is probably never what you want 😄 
That's why Loupe does not allow a value higher than `1000` at the moment.

I think it's better to force users to think about a limit rather than building an application that would work without a limit. Again, no application can work without a limit, it's a logical problem, not a Loupe configuration problem.

By the way: MeiliSearch also has that limit https://www.meilisearch.com/docs/reference/api/settings#body-5

**Comment by alexander-schranz** [comment]
I understand the point that there is always a limit and not a null or PHP_INT_MAX used, for my case the issue is more about support for `offset`.

The offset limit has other advantages you only can first show `2` elements and then on a more load button load the next 10 elements (offset: 2, limit: 10) this is with a normal pagination not possible.

**Comment by Toflar** [comment]
Yeah I don't want to eliminate the possiblity of having `offset` and `limit` making it into Loupe. I just don't need it at the moment and anybody is free to work on that. I think we have to pay extra attention to what happens if it's used in combination with `hitsPerPage` and `page`. I know MeiliSearch also allows both so one just has to take precedence or we have to throw an exception. Not sure yet :)
